combat-trainer.lic

```diff
@@ -1158,6 +1205,94 @@ class SpellProcess
   end
 end
 
+class PetProcess
```

Comment:
This is the heart of the PR. PetProcess is not ZombieProcess because necromancers can use other pets like mudmen, and are supposedly getting a permanent risen pet. My hope is that this process would handle mudman and risen, or anything else.

Zombies can be buffed with the RPU spell, but it does not show up in active spells. You have to look at the zombie to see if the buff spell is active. I think this process would be a good place to expand on that.

Reply:
Also, zombies can perform a 'special' attack if commanded. It was suggested that we detect if a zombie is advancing on something and then issue the the special attack command. I think this process would be a good place to handle that.

Reply:
Could you take a look at the empath pet logic and see if it would be easy to move into this Process? I don't think we need to actually move it now, but I want to make sure this is flexible enough (which it looks like it is).

Reply:
Yes, there is a command issued to a guardian spirit after the spell is casted. That could be moved in to PetProcess in some new method `command_spirit`, where we check if the spell is active before issuing the command. I don't have an empath with the Guardian Spirit spell else I could offer more.

Reply:
Sounds good to me